test/e2e/suites/base/schedule_multi_template_test.go

```diff
@@ -0,0 +1,259 @@
+						gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+					}
+				})
+			})
+
+			ginkgo.By("create PropagationPolicy to propagate FlinkDeployment", func() {
+				pp := testhelper.NewPropagationPolicy(flinkDeploymentNamespace, flinkDeploymentName, []policyv1alpha1.ResourceSelector{
+					{
+						APIVersion: flinkDeploymentObj.GetAPIVersion(),
+						Kind:       flinkDeploymentObj.GetKind(),
+						Name:       flinkDeploymentObj.GetName(),
+					},
+				}, policyv1alpha1.Placement{
+					ClusterAffinity: &policyv1alpha1.ClusterAffinity{
+						ClusterNames: framework.ClusterNames(),
+					},
+					SpreadConstraints: []policyv1alpha1.SpreadConstraint{
+						{
+							SpreadByField: policyv1alpha1.SpreadByFieldCluster,
+							MaxGroups:     1,
+							MinGroups:     1,
+						},
+					},
+					ReplicaScheduling: &policyv1alpha1.ReplicaSchedulingStrategy{
+						ReplicaSchedulingType:     policyv1alpha1.ReplicaSchedulingTypeDivided,
+						ReplicaDivisionPreference: policyv1alpha1.ReplicaDivisionPreferenceAggregated,
+					},
+				})
+				framework.CreatePropagationPolicy(karmadaClient, pp)
+				ginkgo.DeferCleanup(func() {
+					framework.RemovePropagationPolicy(karmadaClient, pp.Namespace, pp.Name)
+				})
+			})
+
+			bindingName := names.GenerateBindingName(flinkDeploymentObj.GetKind(), flinkDeploymentObj.GetName())
+			var resourceBinding *workv1alpha2.ResourceBinding
+
+			// Helper function to get ResourceBinding
+			getResourceBinding := func() (*workv1alpha2.ResourceBinding, error) {
+				return karmadaClient.WorkV1alpha2().ResourceBindings(flinkDeploymentNamespace).Get(context.Background(), bindingName, metav1.GetOptions{})
+			}
+
+			// Helper function to find component by name
+			findComponent := func(components []workv1alpha2.Component, name string) *workv1alpha2.Component {
+				for i := range components {
+					if components[i].Name == name {
+						return &components[i]
+					}
+				}
+				return nil
+			}
+
+			// Helper function to verify component resource requirements
+			verifyComponentResources := func(component *workv1alpha2.Component, expectedReplicas int32, expectedCPU, expectedMemory string) {
+				gomega.Expect(component).ShouldNot(gomega.BeNil())
+				gomega.Expect(component.Replicas).Should(gomega.Equal(expectedReplicas))
+				gomega.Expect(component.ReplicaRequirements).ShouldNot(gomega.BeNil())
+				gomega.Expect(component.ReplicaRequirements.ResourceRequest).Should(gomega.Equal(corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse(expectedCPU),
+					corev1.ResourceMemory: resource.MustParse(expectedMemory),
+				}))
+			}
+
+			ginkgo.By("verify ResourceBinding components field", func() {
+				var jobManagerComponent, taskManagerComponent *workv1alpha2.Component
+
+				// Wait for ResourceBinding to be created and components field to be populated
+				gomega.Eventually(func(g gomega.Gomega) {
+					var err error
+					resourceBinding, err = getResourceBinding()
```

Comment:
```suggestion
					resourceBinding, err = karmadaClient.WorkV1alpha2().ResourceBindings(flinkDeploymentNamespace).Get(context.Background(), bindingName, metav1.GetOptions{})
```
I think we can inline this `one-line` function.